include/grpc/grpc_security.h

```diff
@@ -142,6 +143,13 @@ typedef struct {
   const char *cert_chain;
 } grpc_ssl_pem_key_cert_pair;
 
+typedef struct {
+    int (*verify_peer_callback)(const char*, const char*, void*);
+    void *verify_peer_callback_userdata;
+    void (*verify_peer_destruct)(void*);
+    bool skip_hostname_verification;
```

Comment:
@justinburke FYI.
It doesn't look like this is consistent to what we discussed here: https://github.com/grpc/grpc/issues/10721
Am I missing something?

Reply:
Yep, you're right. First pass at implementing this we were aiming to drop in identical code logic that we had in Java where specifying a custom hostname verifier completely replaces the original hostname verification.

But I'll concede that's a separate (and possibly undesirable) concern. Let me take another pass at this PR with the skip option pulled out.